dev/maven-plugin/src/main/scala/com/lightbend/lagom/maven/MavenFacade.scala

```diff
@@ -0,0 +1,276 @@
+package com.lightbend.lagom.maven
+
+import java.io.File
+import java.util
+import java.util.Collections
+import javax.inject.{ Inject, Singleton }
+
+import org.apache.maven.RepositoryUtils
+import org.apache.maven.artifact.ArtifactUtils
+import org.apache.maven.execution.MavenSession
+import org.apache.maven.lifecycle.internal._
+import org.apache.maven.model.Plugin
+import org.apache.maven.plugin.{ BuildPluginManager, MojoExecution }
+import org.apache.maven.project.MavenProject
+import org.codehaus.plexus.configuration.PlexusConfiguration
+import org.codehaus.plexus.util.StringUtils
+import org.codehaus.plexus.util.xml.Xpp3Dom
+import org.eclipse.aether.{ DefaultRepositorySystemSession, RepositorySystem }
+import org.eclipse.aether.artifact.Artifact
+import org.eclipse.aether.collection.CollectRequest
+import org.eclipse.aether.graph.Dependency
+import org.eclipse.aether.repository.{ WorkspaceReader, WorkspaceRepository }
+import org.eclipse.aether.resolution.{ DependencyRequest, DependencyResult }
+
+import scala.collection.JavaConverters._
+
+/**
+ * Facade in front of maven.
+ *
+ * All the hairy stuff goes here.
+ */
+@Singleton
+class MavenFacade @Inject() (repoSystem: RepositorySystem, session: MavenSession,
+    buildPluginManager: BuildPluginManager, lifecycleExecutionPlanCalculator: LifecycleExecutionPlanCalculator) {
+
+  val cachedProjectIndex = new ProjectIndex(session.getProjects)
+
+  /**
+   * Resolve the classpath for the given artifact.
+   *
+   * @return The classpath.
+   */
+  def resolveArtifact(artifact: Artifact): Seq[Artifact] = {
+    val collect = new CollectRequest()
+    collect.setRoot(new Dependency(artifact, "runtime"))
+    collect.setRepositories(session.getCurrentProject.getRemoteProjectRepositories)
+
+    toDependencies(resolveDependencies(collect)).map(_.getArtifact)
+  }
+
+  /**
+   * Resolve a project, with additional dependencies added to the project.
+   *
+   * @param project The project to resolve.
+   * @param additionalDependencies The additional dependencies to add.
+   * @return The resolved project.
+   */
+  def resolveProject(project: MavenProject, additionalDependencies: Seq[Dependency]): Seq[Dependency] = {
+
+    // We use a low level API rather than just resolving the project so we can inject our own dev mode dependencies
+    // The implementation of this is modelled off org.apache.maven.project.DefaultProjectDependenciesResolver
+    val collect = new CollectRequest()
+    collect.setRootArtifact(RepositoryUtils.toArtifact(project.getArtifact))
+    collect.setRequestContext("dev-mode")
+    collect.setRepositories(project.getRemoteProjectRepositories)
+
+    val stereotypes = session.getRepositorySession.getArtifactTypeRegistry
+
+    // Add project dependencies
+    project.getDependencies.asScala.foreach { dep =>
+      if (!(
+        StringUtils.isEmpty(dep.getGroupId) ||
+        StringUtils.isEmpty(dep.getArtifactId) ||
+        StringUtils.isEmpty(dep.getVersion)
```

Comment:
I'd rather leave as it is - this code was copied/converted to Scala from the class mentioned in the comments above, by leaving it as it is it makes it easier to correlate back to the original source it was copied from in case we have any problems and want to see how Maven does it.